### PR TITLE
Binding portals to one client

### PR DESCRIPTION
**Pull request by danigargar**
#### Type Of Change 
- [ ] Small bug fix
- [x] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, portal/platform, kamusers, agis, ..)
- [x] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

File: schema/DoctrineMigrations/Version20240627075829.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+final class Version20240627075829 extends LoggableMigration
+{
+    public function getDescription(): string
+    {
+        return 'Added company reference to WebPortals';
+    }
+
+    public function up(Schema $schema): void
+    {
+    // this up() migration is auto-generated, please modify it to your needs
```
Comment (rbatistadev) [inline schema/DoctrineMigrations/Version20240627075829.php:22]: remove line

File: schema/DoctrineMigrations/Version20240627075829.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        $this->addSql('ALTER TABLE WebPortals ADD companyId INT UNSIGNED DEFAULT NULL');
+        $this->addSql('ALTER TABLE WebPortals ADD CONSTRAINT FK_C811E30C2480E723 FOREIGN KEY (companyId) REFERENCES Companies (id) ON DELETE CASCADE');
+        $this->addSql('CREATE INDEX IDX_C811E30C2480E723 ON WebPortals (companyId)');
+    }
+
+    public function down(Schema $schema): void
+    {
+        // this down() migration is auto-generated, please modify it to your needs
```
Comment (rbatistadev) [inline schema/DoctrineMigrations/Version20240627075829.php:30]: remove line

File: library/Ivoz/Provider/Domain/Model/WebPortal/WebPortal.php [hunk 3]
```diff
@@ -36,6 +38,19 @@ protected function sanitizeValues(): void
...
+            if ($isUserUrlType && !$this->getCompany()?->isVpbx()) {
+                throw new \DomainException("Company must be vpbx in user urls");
+            }
+
+            $companyBrandId = $this->getCompany()?->getBrand()->getId();
+            $brandId = $this->getBrand()?->getId();
+            if ($companyBrandId !== $brandId) {
+                throw new \DomainException("Company brands must be identical to Brand");
+            }
+        }
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Domain/Model/WebPortal/WebPortal.php:53]:
```suggestion
        // $company = $this->getCompany();
        // if(is_null($company)){
        //    return;
        // }
        
        // Check if the URL type is a user URL
        $isUserUrlType = $this->getUrlType() === self::URLTYPE_USER;

        // If the URL is a user URL and the company is not a VPBX, throw an exception
        if ($isUserUrlType) {
            $company = $this->getCompany();
            if ($company === null || !$company->isVpbx()) {
                throw new \DomainException("Company must be VPBX in user URLs");
            }
        }

        // Retrieve company brand ID and brand ID
        $company = $this->getCompany();
        $companyBrandId = $company 
            ? $company->getBrand()->getId() 
            : null;
        $brandId = $$this->getBrand()?->getId()

        // If the company brand ID does not match the brand ID, throw an exception
        if ($companyBrandId !== $brandId) {
            throw new \DomainException("Company brands must be identical to Brand");
        }
        
```

**Review by rbatistadev — CHANGES_REQUESTED** [review]
.

File: library/Ivoz/Provider/Domain/Model/WebPortal/WebPortal.php (hunk 3 above)
Comment (danigargar) [reply library/Ivoz/Provider/Domain/Model/WebPortal/WebPortal.php:53]: It is not compulsory for a WebPortal to have a company. That's why I introduced the isGeneric() method. Generic is used in front for a WebPortal not binded to a company.

If a portal have not company then the new rules don’t apply at all.

**Review by rbatistadev — APPROVED** [review]

**Pull request merged**